backend/server/routers/programs.py

```diff
@@ -211,4 +212,21 @@ def get_structure(
             add_subgroup_container(structure, "General", container, [])
 
     
-    return {"structure": structure}
\ No newline at end of file
+    return {"structure": structure}
+
+@router.get(
+    "/getGenEds/{programCode}",
+    responses={
+        400: {
+            "description": "The given program code could not be found in the database",
+        },
+        200: {
+            "description": "Returns all geneds available to a given to the given code",
+        },
```

Comment:
Would you mind giving a more concrete example here?

Reply:
ah yes, only didn't cause the course list is so long